app/code/core/Mage/Index/Model/Observer.php

```diff
@@ -136,4 +144,53 @@ public function processConfigDataSave(Varien_Event_Observer $observer)
             Mage_Index_Model_Event::TYPE_SAVE
         );
     }
+
+    /**
+     * Clean old index events for indexers in manual mode
+     *
+     * @return void
+     * @throws Exception
+     */
+    public function cleanOutdatedEvents()
+    {
+        $manualIndexProcessCollection = Mage::getSingleton('index/indexer')
+            ->getProcessesCollection()
+            ->addFieldToFilter('mode', Mage_Index_Model_Process::MODE_MANUAL);
+
+        $now = new DateTime();
+        /** @noinspection PhpUnhandledExceptionInspection */
+        $dateInterval = new DateInterval('PT' . self::OLD_INDEX_EVENT_THRESHOLD_SECONDS . 'S');
+        $oldEventsThreshold = $now
+            ->sub($dateInterval)
+            ->format(Varien_Db_Adapter_Pdo_Mysql::TIMESTAMP_FORMAT);
+
+        $coreResource = Mage::getSingleton('core/resource');
+
+        $writeConnection = $coreResource->getConnection('core_write');
+        $indexEventTableName = $coreResource->getTableName('index/event');
+
+        /** @var Mage_Index_Model_Process $process */
+        foreach ($manualIndexProcessCollection as $process) {
+            $unprocessedEventsCollection = $process
+                ->getUnprocessedEventsCollection()
+                ->addFieldToFilter('created_at', ['lt' => $oldEventsThreshold])
+                ->load();
+
+            $eventList = [];
+            /** @var Mage_Index_Model_Event $unprocessedEvent */
+            foreach ($unprocessedEventsCollection as $unprocessedEvent) {
+                $eventList[] = $unprocessedEvent->getId();
```

Comment:
can we limit it to something like 1000 events? otherwise the "delete where in(...)" could become too long and throw an error